### DISCUSSION TITLE
Crystal shape IPF triangle

### DISCUSSION THREAD

**Discussion by hchaieb1234** [mtex-toolbox/mtex#1289] [Ask Anything] [answered]
Hello,

I need to plot the three edges of my IPF triangle with crystal shapes for better visualization. Could you please tell me how can I achieve this? Thank you!

**Comment by kilir** [comment, marked as answer]
Hi, there are infinite possibilities since in the ipf only one crystal directions needs to be parallel to a specimen direction.
However, here one example:
```
cs = crystalSymmetry('432')
r=xvector;

ori  = [orientation.map(Miller(0,0,1,cs),r) ...
        orientation.map(Miller(0,1,1,cs),r) ...
        orientation.map(Miller(1,1,1,cs),r)]

cS = crystalShape.cube(cs)

plotIPDF(ori, r,'noLabels')
plot(ori(1),ori(1).*cS*0.1,'add2all')
plot(ori(2),ori(2).*cS*0.1,'add2all')
plot(ori(3),ori(3).*cS*0.1,'add2all')
```

Cheers,
Rüdiger
